aten/src/ATen/native/UpSample.h

```diff
@@ -0,0 +1,249 @@
+#include <math.h>
+
+#include <ATen/ATen.h>
+
+namespace at {
+namespace native {
+
+static inline void check_dim_size(
+    const Tensor& data,
+    int64_t dim,
+    int64_t dim_size,
+    int64_t size) {
+  /* Check dimension size of a tensor */
+  AT_CHECK(
+      data.dim() == dim && data.size(dim_size) == size,
+      "Expected tensor of dimension ",
+      dim,
+      " and tensor.size[",
+      dim_size,
+      "] == ",
+      size,
+      " but got: dimension ",
+      data.dim(),
+      " and tensor.size[",
+      dim_size,
+      "] = ",
+      data.size(dim_size));
+}
+
+static inline void upsample_1d_shape_check(
+    const Tensor& data,
+    int64_t type_check,
+    int64_t nbatch,
+    int64_t nchannels,
+    int64_t input_width,
+    int64_t output_width) {
+  AT_CHECK(
+      input_width > 0 && output_width > 0,
+      "Input and output sizes should be greater than 0, but got input (W: ",
+      input_width,
+      ") and output (W: ",
+      output_width,
+      ")");
+
+  if (type_check == 0) {
+    AT_CHECK(
+        !data.numel() == 0 && data.dim() == 3,
+        "Non-empty 3D data tensor expected but got: ",
+        data.dim(),
+        "D");
```

Comment:
This error message is worse. If the tensor is empty, you'll get the message "Non-empty 3D data tensor expected but got: 3D". You should just print the sizes here (they can be passed directly here and will DTRT).

Reply:
right, it makes sense. 

> You should just print the sizes here (they can be passed directly here and will DTRT).

not sure If I am following you in this part.  should I change all the message just for `data.sizes()`

Reply:
and If I split the check, one that will check if it is empty .. and the other one to check if it has the correctly dimension?

Reply:
You should just have the error message say what the size of the tensor we got was. The user will understand. No need to split the check (although, I won't say no if you decide you want to.)

Reply:
sounds good. I am working on that.